tests/queries/1_stateful/00178_random_sample_from_array.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC2154
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+# Initialize variables
+total_tests=0
+passed_tests=0
+
+# Test randomSampleFromArray with integer array
+query_result=$(clickhouse-client -q "SELECT randomSampleFromArray([1,2,3], 2)")
+if [[ "$query_result" =~ "1" ]] || [[ "$query_result" =~ "2" ]] || [[ "$query_result" =~ "3" ]]; then
+    echo "Integer Test: Passed"
+    ((passed_tests++))
+else
+    echo "Integer Test: Failed"
+    echo "Output: $query_result"
+fi
+((total_tests++))
+
+# Test randomSampleFromArray with string array
+query_result=$(clickhouse-client -q "SELECT randomSampleFromArray(['a','b','c'], 2)")
+if [[ "$query_result" =~ "a" ]] || [[ "$query_result" =~ "b" ]] || [[ "$query_result" =~ "c" ]]; then
+    echo "String Test: Passed"
+    ((passed_tests++))
+else
+    echo "String Test: Failed"
+    echo "Output: $query_result"
+fi
+((total_tests++))
+
+# Test randomSampleFromArray with nested arrays
+query_result=$(clickhouse-client -q "SELECT randomSampleFromArray([[1,2],[3,4],[5,6]], 2)")
+if [[ "$query_result" =~ "[1,2]" ]] || [[ "$query_result" =~ "[3,4]" ]] || [[ "$query_result" =~ "[5,6]" ]]; then
+    echo "Nested Array Test: Passed"
+    ((passed_tests++))
+else
+    echo "Nested Array Test: Failed"
+    echo "Output: $query_result"
+fi
+((total_tests++))
+
+# Test with higher sample number than array length
+query_result=$(clickhouse-client -q "SELECT randomSampleFromArray([1,2,3], 5)")
+sorted_result=($(echo "$query_result" | tr -d '[]' | tr ',' '\n' | sort -n))
+sorted_original=("1" "2" "3")
+if [[ "${sorted_result[@]}" == "${sorted_original[@]}" ]]; then
+    echo "Higher Sample Number Test: Passed"
+    ((passed_tests++))
+else
+    echo "Higher Sample Number Test: Failed"
+    echo "query_result: $query_result"
+    echo "sorted_result: ${sorted_result[*]}"
+fi
+((total_tests++))
+
```

Comment:
You might want to include tests for the cases that the array or the sample count are non-const. See SQL function `materialize()` for that.

(To simply things, I think it makes sense to require that the sample count argument is const. You can check that in getReturnTypeImpl().)

Reply:
No problem, I will add a test for non-const samples.
I already added this check:
```
if (!isUnsignedInteger(arguments[1]))
            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Second argument must be a unsigned integer");
```

Reply:
Yes. But not only check but throw an exception if it isn't. Non-const `samples` are an edge case which complicate the implementation.

Reply:
How to solve the problem for the array param?

Reply:
Take a look at the implementation of `arrayRotateLeft`. The code needs to be able to handle const and non-const array parameters.